pandas/core/indexes/interval.py

```diff
@@ -1006,7 +1006,7 @@ def _intersection(self, other, sort):
             # duplicates
             taken = self._intersection_non_unique(other)
 
-        if sort is None:
+        if sort:
```

Comment:
This looks like an actual behavior change? Unless we never actually hit here with a truthy `sort`.

Reply:
AFAICT based on the tests and the type annotations this should not be `None`, only `: bool`. Not sure if that's totally intentional though...

Reply:
This goes back to #42293. On the base implementation, `sort=None` was handled:

https://github.com/pandas-dev/pandas/blob/17ecb56c219d8d01399faffcffa8f059ecf6fd45/pandas/core/indexes/base.py#L6587

Now `sort=None` is still allowed, but it seems to expect only True/False:

https://github.com/pandas-dev/pandas/blob/97c4ce390ae7dafb7f58a21184cc32d921edb6c6/pandas/core/indexes/base.py#L7659`

From this, I think it's just some residual `None` logic that needs a bit more cleaning.